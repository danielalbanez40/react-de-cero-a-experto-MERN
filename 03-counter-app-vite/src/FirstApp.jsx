import React from 'react';
import PropTypes from 'prop-types';

const FirstApp = ({ title, subtitle='La tierra explota!', name }) => {

  return (
    <>
      <h1 data-testid="test-title">{title}</h1>
      {/* <code>{JSON.stringify(newMessage)}</code> */}
      <p>{subtitle}</p>
      <p>{subtitle}</p>
      <p>{name}</p>
    </>
  );
};

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string.isRequired
}

FirstApp.defaultProps = {
  title: 'No hay Título',
  subtitle: 'No hay subtítulo',
  name: 'Daniel Albanez'
}

export default FirstApp;