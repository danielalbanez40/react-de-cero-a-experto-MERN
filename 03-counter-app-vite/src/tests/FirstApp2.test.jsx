import { render, screen } from '@testing-library/react';
import FirstApp from '../FirstApp';

describe('Tests in <FirstApp/>', () => {
  const title = 'Hola, Soy Goku';
  const subtitle = 'Soy un Subtitulo';

  test('should make with snapshot', () => {
    const { container } = render(<FirstApp title={title} />);
    expect(container).toMatchSnapshot();
  });

  test('should show message "Hola Soy Goku"', () => {
    render(<FirstApp title={title} />);
    // screen.debug();
    expect(screen.getByText(title)).toBeTruthy();
  });

  test('should show title in <h1> tag', () => {
    render(<FirstApp title={title} />);
    expect(screen.getByRole('heading', { level: 1 }).innerHTML).toContain(title);
  });
  
  test('should show the subtitle send from props', () => {
    render(<FirstApp title={title} subtitle={subtitle} />);
    expect(screen.getAllByText(subtitle).length).toBe(2);
  });
});
