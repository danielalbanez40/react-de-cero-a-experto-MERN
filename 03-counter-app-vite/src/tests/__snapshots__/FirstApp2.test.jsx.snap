// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <FirstApp/> should make with snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola, Soy Goku
  </h1>
  <p>
    No hay subtítulo
  </p>
  <p>
    No hay subtítulo
  </p>
  <p>
    Daniel Albanez
  </p>
</div>
`;
